auth.go

```diff
@@ -410,16 +422,43 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 // The internal session expiry is controlled with the member NewSessionExpiresAfter.
 // The session key is typically sent to the client as a cookie.
 func (x *Central) Login(identity, password string) (sessionkey string, token *Token, err error) {
-	token = &Token{}
-	if err = x.authenticator.Authenticate(identity, password); err != nil {
-		x.Stats.IncrementInvalidPasswords(x.Log)
-		x.Log.Infof("Login Authentication failed (%v) (%v)", identity, err)
-		return sessionkey, token, err
-	}
 	var userId UserId
 	userId, err = x.userStore.GetUserIdFromIdentity(identity)
 	if err != nil {
 		x.Log.Errorf("UserId could not be found for identity (%v), (%v)", identity, err)
+		return sessionkey, token, ErrIdentityAuthNotFound
+	}
+
+	user, err := x.userStore.GetUser(userId)
+	if err != nil {
+		return sessionkey, token, err
+	}
+
+	token = &Token{}
+	if user.Type == LDAPUserType {
+		err = x.ldap.Authenticate(identity, password)
+		// We want to return Invalid Password or IdentityAuthNotFound, not Invalid Credentials
+		// as LDAP doesnt differentiate between the 2
+		if err == ErrInvalidCredentials {
+			if identity != user.Username {
+				// If the user logged in with his email address, we would find a userid,
+				// but the authentication would fail, we need to check if he logged in with his
+				// username, if he did, then we can just return invalid password, else we return
+				// identity not found
+				err = ErrIdentityAuthNotFound
```

Comment:
This logic doesn't make sense:
1. We successfully found a user in our system for this "identity"
2. We claim that this identity is not found
Something is wrong with this picture.

I don't see the point in disallowing a login because the user entered his email address instead of his username.